[Failing Test]: Spanner change stream tests failing on latest GCP BOM

### What happened?

see https://github.com/apache/beam/runs/48025477476 https://github.com/apache/beam/runs/48411385440

error message:

no result returned
```
java.lang.AssertionError: ParDo(ModsToString)/ParMultiDo(ModsToString).output: 
Expected: iterable with items [...] in any order
but: no item matches: "..." in []
	at org.apache.beam.sdk.testing.PAssert$PAssertionSite.capture(PAssert.java:176)
	at org.apache.beam.sdk.testing.PAssert.that(PAssert.java:461)
	at org.apache.beam.sdk.testing.PAssert.that(PAssert.java:453)
	at org.apache.beam.sdk.io.gcp.spanner.changestreams.it.SpannerChangeStreamIT.testReadSpannerChangeStreamImpl(SpannerChangeStreamIT.java:162)
```

```
java.lang.AssertionError: 
Expected: (stack trace contains string 'SpannerException' and stack trace contains string 'NOT_FOUND: Session not found')
     but: stack trace contains string 'NOT_FOUND: Session not found' was <org.apache.beam.sdk.Pipeline$PipelineExecutionException: com.google.cloud.spanner.SpannerException: INVALID_ARGUMENT: com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Multiplexed sessions may not be deleted>
Stacktrace was: org.apache.beam.sdk.Pipeline$PipelineExecutionException: com.google.cloud.spanner.SpannerException: INVALID_ARGUMENT: com.google.api.gax.rpc.InvalidArgumentException: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Multiplexed sessions may not be deleted
```

and other errors.

this is blocking GCP BOM upgrade #35854.

### Issue Failure

Failure: Test is flaky

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [x] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
+++ b/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy
@@ -612,7 +612,7 @@ class BeamModulePlugin implements Plugin<Project> {
     def google_ads_version = "33.0.0"
     def google_clients_version = "2.0.0"
     def google_cloud_bigdataoss_version = "2.2.26"
-    // [bomupgrader] TODO(#35868): currently pinned, should be determined by: com.google.cloud:google-cloud-spanner, consistent with: google_cloud_platform_libraries_bom
+    // [bomupgrader] TODO(#37008): currently pinned before gcp-bom moving beyond this version, should be determined by: com.google.cloud:google-cloud-spanner, consistent with: google_cloud_platform_libraries_bom
     def google_cloud_spanner_version = "6.104.0"
     def google_code_gson_version = "2.10.1"
     def google_oauth_clients_version = "1.34.1"

